### PR TITLE
chore: Disable default golangci-lint exclusions

### PR DESCRIPTION
**Pull request by killianmuldoon**
Set `.issues.exclude-use-default: false` to expose new linter findings and fix them. 

This will enable checking for package comments and comments on exported types going forward.

### CODE REVIEW

File: api/v1alpha1/hostdevicenetwork_types.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 )
 
 const (
+	// HostDeviceNetworkCRDName is used for the CRD Kind.
```
Comment (killianmuldoon) [inline api/v1alpha1/hostdevicenetwork_types.go:24]: Most of the changes in this PR are adding comments or slightly changing existing comments to meet Go conventions.

File: hack/release.go [hunk 2]
```diff
@@ -133,7 +135,7 @@ func main() {
 		}
 		// Generate new file full path
 		outputFile := filepath.Join(*outputDir, strings.Replace(filepath.Base(file), ".template", ".yaml", 1))
-		f, err := os.Create(outputFile)
+		f, err := os.Create(filepath.Clean(outputFile))
```
Comment (killianmuldoon) [inline hack/release.go:138]: There's three instances of adding filepath.Clean()  to file operations which was caught by the `gosec` linter.

File: api/v1alpha1/hostdevicenetwork_types.go (hunk 1 above)
Comment (killianmuldoon) [reply api/v1alpha1/hostdevicenetwork_types.go:24]: Enabling this now and fixing outstanding issues means we'll have a check for comments on packages and exported types in future.

**Comment by rollandf** [comment]
Hey, add a new line to the PR text to make `Conventional Commit PR message check / check-for-cc (pull_request)` happy

**Comment by almaslennikov** [comment]
Hey, after merging https://github.com/Mellanox/network-operator/pull/706 the project layout has changed. Could you rebase your PR?

**Review by rollandf — APPROVED** [review]

File: .golangci.yml [hunk 3]
```diff
@@ -87,6 +87,12 @@ linters:
     - whitespace
 
 issues:
+  # Disable default golangci exclusions.
+  exclude-use-default: false
```
[the remark this answers is not in the document]
Comment (killianmuldoon) [reply .golangci.yml:91]: IMO yes - especially to enable linting for correct comments on exported types and packages. Having those comments will let us improve our godocs over time 

That's the vast majority of issues that were caught here - with a couple of additional things from gosec.

**Review by e0ne — APPROVED** [review]

**Pull request merged**